source/dialogs/combi.c

```diff
@@ -150,41 +150,87 @@ static void combi_mode_destroy ( Mode *sw )
         mode_set_private_data ( sw, NULL );
     }
 }
+
+static void split_bang( const char *input, char **bang, char **rest ) {
+	// Splits string input into a part containing the bang and the part containing the rest,
+	// saved in the pointers bang and rest.
+    if ( input != NULL) {
```

Comment:
I prefer to avoid the use of goto.

Reply:
Setting `*bang = NULL` means I need to do checks afterward, so for now, I left it as an empty string. I did however rework the function with a lot of early returns.

Reply:
Sure, you change `strlen(bang) > 0` to `bang != NULL`, it’s that big of a change, is it? :-)

Reply:
Fair enough, changed that now. Code got slightly more complicated because I found a bug: I was only looking for the first '!', while in fact, the first one could be part of a matching token, so now, we’re looking for the first one that is preceded by a space.